maas-region/src/charm.py

```diff
@@ -544,6 +551,26 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent):
             self._update_tls_config()
             self._update_prometheus_config(self.config["enable_prometheus_metrics"])  # type: ignore
 
+    def _on_msm_created(self, event: ops.RelationCreatedEvent) -> None:
+        logger.info(event)
+        if self.unit.is_leader():
+            if cluster_uuid := MaasHelper.get_maas_uuid():
+                self.msm.request_enrol(cluster_uuid)
+            else:
+                event.defer()
+
+    def _on_msm_removed(self, event: enrol.TokenWithdrawEvent) -> None:
+        logger.info(event)
```

Comment:
could we add explanation why this is noop or why we are not calling `msm withdraw`? It will be a good reference for future readers

Reply:
done